src/som/interpreter/Interpreter.java

```diff
@@ -325,39 +329,22 @@ public SAbstractObject start() throws ProgramDefinitionError {
     }
   }
 
-  public Frame pushNewFrame(final SMethod method, final Frame contextFrame) {
-    // Allocate a new frame and make it the current one
-    frame = universe.newFrame(frame, method, contextFrame);
-
-    // Return the freshly allocated and pushed frame
-    return frame;
-  }
-
-  public Frame pushNewFrame(final SMethod method) {
-    return pushNewFrame(method, null);
-  }
-
-  public Frame getFrame() {
-    // Get the frame from the interpreter
-    return frame;
-  }
-
-  public SMethod getMethod() {
+  public SMethod getMethod(Frame frame) {
     // Get the method from the interpreter
-    return getFrame().getMethod();
+    return frame.getMethod();
   }
 
-  public SAbstractObject getSelf() {
+  public SAbstractObject getSelf(Frame frame) {
     // Get the self object from the interpreter
-    return getFrame().getOuterContext(universe.nilObject).getArgument(0, 0);
+    return frame.getOuterContext(universe.nilObject).getArgument(0, 0);
   }
 
   private void send(final SSymbol selector, final SClass receiverClass,
-      final int bytecodeIndex) {
+      final int bytecodeIndex, Frame frame) throws ReturnException {
```

Comment:
Hm, you don't throw a `ReturnException` here, do you?

Reply:
True, removed.